pkg/master/thirdparty_controller.go

```diff
@@ -37,8 +37,21 @@ func makeThirdPartyPath(group string) string {
 	return thirdpartyprefix + "/" + group
 }
 
-func getThirdPartyGroupName(path string) string {
-	return strings.TrimPrefix(strings.TrimPrefix(path, thirdpartyprefix), "/")
+func makeThirdPartyVersionedPath(group, version string) string {
+	if len(group) == 0 {
+		return thirdpartyprefix
+	}
+	return fmt.Sprintf("%s/%s/%s", thirdpartyprefix, group, version)
+}
+
+func getThirdPartyGroupKind(path string) (string, string) {
```

Comment:
Generally speaking, in kubernetes codebase, we use kind for the Go type (ex: "ReplicationController") and resource name for the name in the URL ("replicationcontrollers"). As per that, it seems like you are extracting resource name here.

Looking at the third party code around, it seems that we are assuming that string.ToLower(kind) + "s" gives us the resource name. We should have had a function for that rather than putting that everywhere.